.vscode/tasks.json

```diff
@@ -0,0 +1,147 @@
+      "type": "shell",
+      "command": "pnpm run build --filter=${input:workspaceName}",
+      "detail": "Build specific package or application"
+    },
+    {
+      "label": "Run tests",
+      "type": "shell",
+      "command": "pnpm run test",
+      "detail": "Run all tests without filter"
+    },
+    {
+      "label": "Run tests for a specific component",
+      "type": "shell",
+      "command": "pnpm run test ${input:componentName}",
+      "detail": "Run all tests for a specific component"
+    },
+    {
+      "label": "Open development servers",
+      "type": "shell",
+      "command": "pnpm dev",
+      "detail": "Open all development servers"
+    },
+    {
+      "label": "Run tests with filter",
+      "type": "shell",
+      "command": "pnpm run test --filter=${input:testFilter}",
+      "detail": "Run different test types"
+    }
+  ],
+  "inputs": [
+    {
+      "type": "pickString",
+      "id": "workspaceName",
+      "description": "Choose a worskspace",
+      "options": [
+        "css-workshop",
+        "react-workshop",
+        "portal",
+        "@itwin/itwinui-css",
+        "@itwin/itwinui-react",
+        "@itwin/itwinui-variables"
+      ]
+    },
+    {
+      "type": "pickString",
+      "id": "componentName",
+      "description": "Choose a component",
+      "options": [
+        "Alert",
+        "Avatar",
+        "AvatarGroup",
+        "Backdrop",
+        "Badge",
+        "Breadcrumbs",
+        "ButtonGroup",
+        "Buttons",
+        "Carousel",
+        "Checkbox",
+        "ColorPicker",
+        "ComboBox",
+        "DatePicker",
+        "Dialog",
+        "Divider",
+        "DropdownMenu",
+        "ExpandableBlock",
+        "Fieldset",
+        "FileUpload",
+        "Flex",
+        "Footer",
+        "Header",
+        "Icon",
+        "InformationPanel",
+        "Input",
+        "InputGrid",
+        "InputGroup",
+        "InputWithDecorations",
+        "Label",
+        "LabeledInput",
+        "LabeledSelect",
+        "LabeledTextarea",
+        "LinkAction",
+        "List",
+        "Menu",
+        "Modal",
+        "NonIdealState",
+        "NotificationMarker",
+        "Overlay",
+        "Popover",
+        "ProgressIndicators",
+        "Radio",
+        "RadioTiles",
+        "SearchBox",
+        "Select",
+        "SideNavigation",
+        "SkipToContentLink",
+        "Slider",
+        "StatusMessage",
+        "Stepper",
+        "Surface",
+        "Table",
+        "Tabs",
+        "Tag",
+        "Textarea",
+        "ThemeProvider",
+        "Tile",
+        "TimePicker",
+        "Toast",
+        "ToggleSwitch",
+        "Tooltip",
+        "TransferList",
+        "Tree",
+        "Typography",
+        "VisuallyHidden"
+      ]
+    },
+    {
+      "type": "pickString",
+      "id": "testFilter",
+      "description": "Choose a test type",
+      "options": [
+        "css-workshop",
+        "react-workshop",
+        "@itwin/itwinui-css",
```

Comment:
I believe `"@itwin/itwinui-css"` can be excluded from this list, because it doesn't have any tests.

Also, I don't know if it's possible but it would be nice if we could associate custom labels with the options. For example, `"css-workshop"` and `"react-workshop"` would be more clearly described as "css-workshop (visual tests)" and "react-workshop (visual tests)".

Reply:
I have found a way to add labels yet so in the meantime, I would include the description between the parentheses.

Reply:
> (Edited) I haven't found a way to add labels yet so in the meantime, I would include the description between the parentheses.

Sorry, I think I made a typo yesterday. I meant I haven't found a way to add labels yet so I might have to just go with parentheses for now.

Reply:
This doesn't work, because the parentheticals will be passed verbatim into the terminal command.

![](https://github.com/user-attachments/assets/2ccd212d-cb2b-458f-919a-2e47dc767211)

Reply:
This doesn't work, because the parentheticals will be passed verbatim into the terminal command.

![](https://github.com/user-attachments/assets/2ccd212d-cb2b-458f-919a-2e47dc767211)

---

If you look at the [docs for input variables](https://code.visualstudio.com/docs/editor/variables-reference#_input-variables), it states this:

> An option can be a string value or an object with both a label and value. The dropdown will display label: value.

So I believe you should be able something like this:

```json
      "options": [
        { "label": "visual tests", "value": "css-workshop" },
        { "label": "visual tests", "value": "react-workshop" },
        { "label": "unit tests", "value": "@itwin/itwinui-react" },
        { "label": "end-to-end tests", "value": "e2e" },
        { "label": "accessibility tests", "value": "a11y" }
      ]
```

(feel free to copy paste this)

Reply:
Oh, I totally missed that while going through the doc. I have made the change.